### ISSUE TITLE
Travis reporting success without actually running the tests

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#16624]
My PR #16603 is currently running tests on Travis:
[https://travis-ci.org/sympy/sympy/builds/518524533?utm_source=github_status&utm_medium=notification](https://travis-ci.org/sympy/sympy/builds/518524533?utm_source=github_status&utm_medium=notification)

Travis is reporting that the tests have passed when I know they should not. The non-doctest jobs that actually run the tests are exiting after around 1 minute erroneously reporting success.

Example test log:
```
Worker information
hostname: 475c6eba-373f-496a-8778-391171466011@1.production-1-worker-org-gce-dnjh
version: v6.2.0 https://github.com/travis-ci/worker/tree/5e5476e01646095f48eec13196fdb3faf8f5cbf7
instance: travis-job-f3cff8ef-3d0a-43df-916b-354c062c8cd5 travis-ci-sardonyx-xenial-1547455603-2c98a19 (via amqp)
startup: 6.314433665s
system_info
Build system information
Build language: python
Build group: stable
Build dist: xenial
Build id: 518524533
Job id: 518524537
Runtime kernel version: 4.15.0-1026-gcp
travis-build version: 9b2b201a2
Build image provisioning date and time
Mon Jan 14 09:10:34 UTC 2019
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
Systemd Version
systemd 229
Cookbooks Version
f9b6c9e https://github.com/travis-ci/travis-cookbooks/tree/f9b6c9e
git version
git version 2.20.1
bash version
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
gcc version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
docker version
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:11:02 2018
 OS/Arch:           linux/amd64
 Experimental:      false
Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:09:05 2018
  OS/Arch:          linux/amd64
  Experimental:     false
clang version
clang version 7.0.0 (tags/RELEASE_700/final)
jq version
jq-1.5
bats version
Bats 0.4.0
shellcheck version
0.6.0
shfmt version
v2.6.2
ccache version
3.2.4
cmake version
cmake version 3.12.4
heroku version
heroku/7.19.4 linux-x64 node-v11.3.0
imagemagick version
Version: ImageMagick 6.8.9-9 Q16 x86_64 2018-09-28 http://www.imagemagick.org
md5deep version
4.4
mercurial version
version 4.8
mysql version
mysql  Ver 14.14 Distrib 5.7.24, for Linux (x86_64) using  EditLine wrapper
openssl version
OpenSSL 1.0.2g  1 Mar 2016
packer version
1.3.3
postgresql client version
psql (PostgreSQL) 10.6 (Ubuntu 10.6-1.pgdg16.04+1)
ragel version
Ragel State Machine Compiler version 6.8 Feb 2013
sudo version
1.8.16
gzip version
gzip 1.6
zip version
Zip 3.0
vim version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
iptables version
iptables v1.6.0
curl version
curl 7.47.0 (x86_64-pc-linux-gnu) libcurl/7.47.0 GnuTLS/3.4.10 zlib/1.2.8 libidn/1.32 librtmp/2.3
wget version
GNU Wget 1.17.1 built on linux-gnu.
rsync version
rsync  version 3.1.1  protocol version 31
gimme version
v1.5.3
nvm version
0.34.0
perlbrew version
/home/travis/perl5/perlbrew/bin/perlbrew  - App::perlbrew/0.85
phpenv version
rbenv 1.1.1-39-g59785f6
rvm version
rvm 1.29.7 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]
default ruby version
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
CouchDB version
couchdb 1.6.1
ElasticSearch version
5.5.0
Installed Firefox version
firefox 63.0.1
MongoDB version
MongoDB 4.0.5
PhantomJS version
2.1.1
Pre-installed PostgreSQL versions
9.4.20
9.5.15
9.6.11
Redis version
redis-server 5.0.3
Pre-installed Go versions
1.11.1
mvn version
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z)
gradle version
Gradle 4.10.2!
lein version
Leiningen 2.8.3 on Java 11.0.1 OpenJDK 64-Bit Server VM
Pre-installed Node.js versions
v10.15.0
v11.0.0
v4.9.1
v6.16.0
v8.12.0
v8.15.0
v8.9
phpenv versions
  system
  5.6
  5.6.36
  7.1.19
* 7.2.7 (set by /home/travis/.phpenv/version)
  hhvm
  hhvm-stable
composer --version
Composer version 1.8.0 2018-12-03 10:31:16
Pre-installed Ruby versions
ruby-2.3.8
ruby-2.4.5
ruby-2.5.3
git.checkout
9.88s$ git clone --depth=50 https://github.com/sympy/sympy.git sympy/sympy
Cloning into 'sympy/sympy'...
remote: Enumerating objects: 60468, done.
remote: Counting objects: 100% (60468/60468), done.
remote: Compressing objects: 100% (20895/20895), done.
remote: Total 60468 (delta 47615), reused 50908 (delta 39470), pack-reused 0
Receiving objects: 100% (60468/60468), 29.36 MiB | 20.89 MiB/s, done.
Resolving deltas: 100% (47615/47615), done.
$ cd sympy/sympy
1.14s$ git fetch origin +refs/pull/16603/merge:
remote: Enumerating objects: 1361, done.
remote: Counting objects: 100% (940/940), done.
remote: Compressing objects: 100% (174/174), done.
remote: Total 512 (delta 430), reused 412 (delta 338), pack-reused 0
Receiving objects: 100% (512/512), 70.17 KiB | 11.70 MiB/s, done.
Resolving deltas: 100% (430/430), completed with 119 local objects.
From https://github.com/sympy/sympy
 * branch              refs/pull/16603/merge -> FETCH_HEAD
$ git checkout -qf FETCH_HEAD
Setting environment variables from .travis.yml
$ export SPLIT="2/2"
$ export TEST_SYMPY="true"
0.01s$ source ~/virtualenv/python3.7/bin/activate
$ python --version
Python 3.7.1
$ pip --version
pip 18.1 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/pip (python 3.7)
before_install.1
0.00s$ set -e
before_install.2
0.02s$ python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
before_install.3
2.30s$ if [[ "${FASTCACHE}" != "false" ]]; then pip install fastcache; fi
Collecting fastcache
  Downloading https://files.pythonhosted.org/packages/fb/98/93f2d36738868e8dd5a8dbfc918169b24658f63e5fa041fe000c22ae4f8b/fastcache-1.0.2.tar.gz
Building wheels for collected packages: fastcache
  Running setup.py bdist_wheel for fastcache ... -done
  Stored in directory: /home/travis/.cache/pip/wheels/b7/90/c0/da92ac52d188d9ebca577044e89a14d0e6ff333c1bcd1ebc14
Successfully built fastcache
Installing collected packages: fastcache
Successfully installed fastcache-1.0.2
before_install.4
0.00s$ if [[ "${TEST_COVERAGE}" == "true" ]]; then pip install coverage; coverage debug sys; fi
before_install.5
1.20s$ if [[ -n "${TEST_OPT_DEPENDENCY}" ]]; then
  # We do this conditionally because it saves us some downloading if the
  # version is the same.
      deactivate; # Deactivate the Travis virtualenv
      if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
        wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -q -O miniconda.sh;
      else
        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -q -O miniconda.sh;
      fi
      bash miniconda.sh -b -p $HOME/miniconda;
      export PATH="$HOME/miniconda/bin:$PATH";
      hash -r;
      conda config --set always_yes yes --set changeps1 no;
      conda update -q conda;
      conda config --prepend channels conda-forge --prepend channels symengine/label/dev;
  
      conda info -a;
      conda create -q -n test-environment ${TEST_OPT_DEPENDENCY};
      source activate test-environment;
      export CPATH=$CONDA_PREFIX/include;
      export LIBRARY_PATH=$CONDA_PREFIX/lib;
      export LD_LIBRARY_PATH=$CONDA_PREFIX/lib;
      export SYMPY_STRICT_COMPILER_CHECKS=1;
      conda clean --all;
      if [[ "$TEST_SAGE" == "true" ]]; then
          # Use a separate environment because sage downgrades matplotlib
          conda create -c conda-forge/label/cf201901 -n sage sagelib mpmath "ipython>=5.5.0,<6";
          conda clean --all;
      fi
      if [[ "$TRAVIS_PYTHON_VERSION" == "3.6" ]]; then
          # Change this back to conda once matchpy 0.5.0 gets updated on conda-forge
          pip install matchpy;
      fi
  elif [ "$TRAVIS_PYTHON_VERSION" != "pypy" ]; then
      pip list | grep "numpy" && pip uninstall -y numpy;
  fi
  
numpy            1.15.4    
Uninstalling numpy-1.15.4:
  Successfully uninstalled numpy-1.15.4
before_install.6
0.00s$ if [[ "${TEST_SPHINX}" == "true" ]]; then pip install "sphinx" "docutils" doctr matplotlib; fi
install.1
0.00s$ set -e
install.2
0.00s$ if [ "$TRAVIS_PYTHON_VERSION" = "pypy" ]; then virtualenv -p /usr/bin/pypy ~/.venv; . ~/.venv/bin/activate; fi
install.3
1.90s$ pip install mpmath
Collecting mpmath
  Downloading https://files.pythonhosted.org/packages/ca/63/3384ebb3b51af9610086b23ea976e6d27d6d97bf140a76a365bd77a3eb32/mpmath-1.1.0.tar.gz (512kB)
Building wheels for collected packages: mpmath
  Running setup.py bdist_wheel for mpmath ... done
  Stored in directory: /home/travis/.cache/pip/wheels/63/9d/8e/37c3f6506ed3f152733a699e92d8e0c9f5e5f01dea262f80ad
Successfully built mpmath
Installing collected packages: mpmath
Successfully installed mpmath-1.1.0
install.4
32.66s$ if [[ "${TEST_SETUP}" == "true" ]]; then
    # The install cycle below is to test installation on systems without
    # setuptools.
    virtualenv ~/.venv-no-setuptools;
    ~/.venv-no-setuptools/bin/pip install mpmath;
    ~/.venv-no-setuptools/bin/pip uninstall -y setuptools;
    ~/.venv-no-setuptools/bin/python -We:invalid setup.py -q install;
  fi
  python -We:invalid -m compileall -f -q sympy/;
  python -We:invalid setup.py -q install;
  pip list --format=columns;
  
/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/setuptools/dist.py:470: UserWarning: Normalizing '1.5.dev' to '1.5.dev0'
  normalized_version,
no previously-included directories found matching 'doc/_build'
warning: no files found matching 'TODO'
Package          Version   
---------------- ----------
atomicwrites     1.2.1     
attrs            18.2.0    
certifi          2018.11.29
fastcache        1.0.2     
mock             2.0.0     
more-itertools   4.3.0     
mpmath           1.1.0     
nose             1.3.7     
pbr              5.1.1     
pip              18.1      
pipenv           2018.11.26
pluggy           0.8.0     
py               1.7.0     
pytest           4.1.1     
setuptools       40.6.3    
six              1.11.0    
sympy            1.5.dev0  
virtualenv       16.1.0    
virtualenv-clone 0.4.0     
wheel            0.32.3    
0.00s$ set -e
The command "set -e" exited with 0.
1.31s$ bin/test_travis.sh
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ -n '' ]]
+mkdir empty
+cd empty
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ -n '' ]]
+[[ '' == *\m\a\t\p\l\o\t\l\i\b* ]]
+[[ '' == *\s\y\m\e\n\g\i\n\e* ]]
+[[ true == \t\r\u\e ]]
+python -We:invalid
+cat
Testing SYMPY, split 2/2
============================= test process starts ==============================
executable:         /home/travis/virtualenv/python3.7.1/bin/python  (3.7.1-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              None
random seed:        73892784
hash randomization: on (PYTHONHASHSEED=1515645635)
split:              2/2
================== tests finished: 0 passed, in 0.00 seconds ===================
+[[ '' == \t\r\u\e ]]
The command "bin/test_travis.sh" exited with 0.
0.00s$ if [[ "${TEST_SPHINX}" == "true" ]]; then doctr deploy dev --deploy-repo sympy/sympy_doc --command './generate_indexes.py'; fi
The command "if [[ "${TEST_SPHINX}" == "true" ]]; then doctr deploy dev --deploy-repo sympy/sympy_doc --command './generate_indexes.py'; fi" exited with 0.
0.00s$ if [[ "${TEST_COVERAGE}" == "true" ]]; then bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"; fi
The command "if [[ "${TEST_COVERAGE}" == "true" ]]; then bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"; fi" exited with 0.
Done. Your build exited with 0.```

**Comment by asmeurer** [comment]
Maybe the environment variables aren't being set correctly.

**Comment by asmeurer** [comment]
I guess the tests are running but it's finding 0 tests. 

What is the first build where this happens?

**Comment by asmeurer** [comment]
I think something in the release branch broke it.

**Comment by asmeurer** [comment]
The issue is from https://github.com/sympy/sympy/pull/16513. It is using `blacklist.extend` instead of `blacklist.append`, so every character in the path is being added to the blacklist.

**Comment by oscarbenjamin** [comment]
Does the blacklist match only the start of a filename? I guess if all tests are in files called `test_` then if `t` is in the black list they all get skipped...

**Comment by asmeurer** [comment]
It does an `in` check, I guess so that you can blacklist entire directories.

**Comment by oscarbenjamin** [comment]
I guess this will only apply to PRs pushed between the merge of 1.4 and when the fix in #16625 is merged.

Presumably it will affect the master branch build for #16586 that @Upabjojr just merged.
